src/app/query/bitcoin/address/utxos-by-address.hooks.ts

```diff
@@ -22,12 +22,56 @@ export function filterUtxosWithInscriptions(
 }
 
 /**
- * Warning: ⚠️ These are **all** UTXOs, including Stamped and Inscribed UTXOs.
- * You should probably use `useCurrentNativeSegwitAccountSpendableUtxos` instead.
+ * Warning: ⚠️ To avoid spending inscriptions, when using UTXOs
+ * we set `filterInscriptionUtxos` and `filterPendingTxsUtxos` to true
  */
-export function useCurrentNativeSegwitUtxos() {
+export function useCurrentNativeSegwitUtxos(
+  { filterInscriptionUtxos, filterPendingTxsUtxos } = {
+    filterInscriptionUtxos: true,
+    filterPendingTxsUtxos: true,
+  }
+) {
   const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
-  return useGetUtxosByAddressQuery(nativeSegwitSigner.address);
+  const address = nativeSegwitSigner.address;
+
+  return useNativeSegwitUtxosByAddress({
+    address,
+    filterInscriptionUtxos,
+    filterPendingTxsUtxos,
+  });
+}
+
+interface UseFilterUtxosByAddressArgs {
+  address: string;
+  filterInscriptionUtxos: boolean;
+  filterPendingTxsUtxos: boolean;
+}
+
+export function useNativeSegwitUtxosByAddress({
+  address,
+  filterInscriptionUtxos,
+  filterPendingTxsUtxos,
+}: UseFilterUtxosByAddressArgs) {
+  const filterOutInscriptions = useFilterInscriptionsByAddress(address);
+  const filterOutPendingTxsUtxos = useFilterPendingUtxosByAddress(address);
+
+  return useGetUtxosByAddressQuery(address, {
+    select(utxos) {
+      if (filterPendingTxsUtxos && filterInscriptionUtxos) {
+        return filterOutPendingTxsUtxos(filterOutInscriptions(utxos));
+      }
+
+      if (filterPendingTxsUtxos) {
+        return filterOutPendingTxsUtxos(utxos);
+      }
+
+      if (filterInscriptionUtxos) {
+        return filterOutInscriptions(utxos);
+      }
+
+      return utxos;
+    },
```

Comment:
How could this be written in less manual way? What if there was a third filter?

Reply:
yeah, don't like this as well, refactored